### PR TITLE
Handle gzip encoding for errors whilst using consume API in BufferFillerActor

### PR DESCRIPTION
**Pull request by nicklester**

### CODE REVIEW

File: server/cmwell-util/src/main/scala/cmwell/util/http/HttpZipDecoder.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package cmwell.util.http
+
+import akka.http.scaladsl.coding.{Deflate, Gzip, NoCoding}
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.HttpEncodings
+
+object HttpZipDecoder {
```
Comment (hochgi) [inline server/cmwell-util/src/main/scala/cmwell/util/http/HttpZipDecoder.scala]: It's not so much a "http util" as a "akka util".
Lets move it to `cmwell.util.akka.http` package

WDYT?

Comment (nicklester) [reply server/cmwell-util/src/main/scala/cmwell/util/http/HttpZipDecoder.scala]: I'm happy @hochgi , good suggestion. Is there anything else?

Comment (nicklester) [reply server/cmwell-util/src/main/scala/cmwell/util/http/HttpZipDecoder.scala]: @hochgi that's done now...

File: server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/Downloader.scala [hunk 2]
```diff
@@ -471,56 +472,59 @@ class Downloader(baseUrl: String,
       val job = Flow[(Seq[Path], State)]
         .map { case (paths, state) => paths -> Some(state) }
         .via(Retry.retryHttp(timeout, parallelism, baseUrl)(createRequest)) // fetch data from paths
+        .map {
```
Comment (e-orz) [inline server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/Downloader.scala]: Decompressing here is too late. The body is read also in the retryHttp and won't be parsed correctly.

File: server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/Downloader.scala [hunk 3]
```diff
@@ -612,14 +616,18 @@ class Downloader(baseUrl: String,
       val job = Flow[(Seq[Uuid], State)]
         .map { case (uuids, state) => uuids -> Some(state) }
         .via(Retry.retryHttp(timeout, parallelism, baseUrl)(createRequest)) // fetch data from uuids
+        .map {
```
Comment (e-orz) [inline server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/Downloader.scala]: Same as above.

**Comment by e-orz** [comment]
Looks good!
Please solve the conflicts and I will approve it.

**Comment by bryaakov** [comment]
Create a new PR without code-styling conflicts: #595

**Pull request closed without merging**